### PR TITLE
bug(fix): PhoneNumberField does not honor readonly prop

### PR DESCRIPTION
**Pull request by joebuono**
#### Description of changes

This PR address [this issue](https://github.com/aws-amplify/amplify-ui/issues/1795): "PhoneNumberField dial code dropdown doesn't honor readonly prop"

As a result of these changes, if the `isReadOnly` prop is passed to the `PhoneNumberField`:
- The `CountryCodeSelect` primitive is marked with the `aria-disabled=“true”` attribute. The reason we’re not marking it with `aria-readonly=“true”` is because that does not cause screen readers to announce anything to customers. @hbuchel 
- All the options in the `CountryCodeSelect` are marked with the `disabled` attribute because the select field should still be readable and focusable, but disallow changing the readonly selection. 
- Form data is still submitted if the `isReadOnly` prop is passed. 

readonlygif

#### Issue #, if available

Fixes https://github.com/aws-amplify/amplify-ui/issues/1795

#### Description of how you validated changes

Ran the docs locally and wrote some unit tests. 

#### Checklist

- [X] PR description included
- [X] `yarn test` passes
- [X] Tests are updated
- [X] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by 0618 — APPROVED** [review]

**Review by jacoblogan — APPROVED** [review]
LGTM

File: packages/react/src/primitives/PhoneNumberField/CountryCodeSelect.tsx [hunk 1]
```diff
@@ -7,20 +7,21 @@ import { CountryCodeSelectProps, Primitive } from '../types';
...
 
 const CountryCodeSelectPrimitive: Primitive<CountryCodeSelectProps, 'select'> =
-  ({ className, dialCodeList, ...props }, ref) => {
+  ({ className, dialCodeList, isReadOnly, ...props }, ref) => {
     const dialList = dialCodeList ?? countryDialCodes;
     const countryCodeOptions = React.useMemo(
       () =>
         dialList.map((dialCode) => (
-          <option key={dialCode} value={dialCode}>
+          <option key={dialCode} value={dialCode} disabled={isReadOnly}>
```
Comment (reesscot) [inline packages/react/src/primitives/PhoneNumberField/CountryCodeSelect.tsx:15]: Should we add a comment of why we are mapping disabled to isReadOnly? We will wonder later on.

File: packages/react/src/primitives/PhoneNumberField/CountryCodeSelect.tsx [hunk 2]
```diff
@@ -7,20 +7,21 @@ import { CountryCodeSelectProps, Primitive } from '../types';
...
             {dialCode}
           </option>
         )),
-      [dialList]
+      [dialList, isReadOnly]
     );
 
     return (
       <SelectField
+        aria-disabled={isReadOnly}
```
Comment (reesscot) [inline packages/react/src/primitives/PhoneNumberField/CountryCodeSelect.tsx:24]: Same here, I think there should be a comment on why this is more accessible.

File: packages/react/src/primitives/PhoneNumberField/__tests__/PhoneNumberField.test.tsx [hunk 3]
```diff
@@ -29,6 +31,34 @@ describe('PhoneNumberField primitive', () => {
     };
   };
 
+  const originalLog = console.log;
```
Comment (reesscot) [inline packages/react/src/primitives/PhoneNumberField/__tests__/PhoneNumberField.test.tsx:34]: nit: put this above the describe, and reassign `originalLog` in an `afterEach`.

File: packages/react/src/primitives/PhoneNumberField/__tests__/PhoneNumberField.test.tsx [hunk 4]
```diff
@@ -135,4 +165,24 @@ describe('PhoneNumberField primitive', () => {
 
     expect(onCountryCodeChange).toHaveBeenCalled();
   });
+
+  it('should set aria-disabled="true" when the isReadOnly prop is passed, and disable all the select options', async () => {
```
Comment (reesscot) [inline packages/react/src/primitives/PhoneNumberField/__tests__/PhoneNumberField.test.tsx:169]: Add comment on why this behavior exists

**Review by reesscot — COMMENTED** [review]
This is looking great, just had some comments. Also, the demo is showing a button that looks too tall. Can we fix that?

**Review by zchenwei — COMMENTED** [review]
Humm, SR wouldn't announce anything from elements marked by `aria-readonly=true`? It feels weird.

**Pull request merged**